### ISSUE TITLE
Unable to load cnn_dailymail

### ISSUE THREAD

**Issue by xuf12** [salesforce/decaNLP#20]
I ran the following command just to train on cnn_dailymail:
```nvidia-docker run -it --rm -v `pwd`:/decaNLP/ -u $(id -u):$(id -g) decanlp bash -c "python /decaNLP/train.py --train_tasks cnn_dailymail --train_iterations 1 --gpu 0"```

and I got the following error message:
```process_main - Constructing field
process_main - Loading cnn_dailymail
process_main - Adding cnn_dailymail to training datasets
Traceback (most recent call last):
  File "/decaNLP/train.py", line 365, in <module>
    main()
  File "/decaNLP/train.py", line 352, in main
    field, train_sets, val_sets = prepare_data(args, field, logger)
  File "/decaNLP/train.py", line 67, in prepare_data
    split = get_splits(args, task, FIELD, **kwargs)[0]
  File "/decaNLP/util.py", line 161, in get_splits
    fields=FIELD, root=args.data, **kwargs)
  File "/decaNLP/text/torchtext/datasets/generic.py", line 426, in splits
    os.path.join(path, 'training.jsonl'), fields, **kwargs)
  File "/decaNLP/text/torchtext/datasets/generic.py", line 353, in __init__
    examples = torch.load(cache_name)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 267, in load
    return _load(f, map_location, pickle_module)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 420, in _load
    result = unpickler.load()
EOFError: Ran out of input
```
I tried running cnn only without any issue. However, when I train for dailymail only, I ran out of RAM (I have 32GB). So the 2 questions I have are:

1. How much RAM do we need to load dailymail alone? 
2. How to load the cnn_dailymail data without EOFError?

Thank you.

**Comment by bmccann** [comment]
There seems to be something in torchtext that blows up the memory requirements. I'm looking into this though. Are you able to run other tasks okay?

**Comment by xuf12** [comment]
Yeah I tried every other task in the round-robin without any issue. I used cnn dailymail instead of cnn-dailymail in the task parameter, and the CNN part worked without any issue.

**Comment by bmccann** [comment]
@xuf12 This was a problem with the memory consumption of the tokenizer we were using (revtok). It was creating too many short strings during tokenization. For now, a quick fix (1f83b7a739502d2d826a0a2a95452a66dbf58273), but we'll get this fixed in revtok.

Let me know if this fixes your issue!

**Comment by bmccann** [comment]
Same problem at https://github.com/salesforce/decaNLP/issues/13. Should be resolved now.
